extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -259,47 +279,94 @@ class AmpBrightcove extends AMP.BaseElement {
   }
 
   /**
-   * @return {string}
+   * Iframe src returns a promise as it waits for consent promises
+   *
+   * @return {!Promise<string>} iframe source url
    * @private
    */
   getIframeSrc_() {
     const {element: el} = this;
+
     const account = userAssert(
       el.getAttribute('data-account'),
       'The data-account attribute is required for <amp-brightcove> %s',
       el
     );
+
     const embed = el.getAttribute('data-embed') || 'default';
 
     this.playerId_ =
       el.getAttribute('data-player') ||
       el.getAttribute('data-player-id') ||
       'default';
 
-    const src =
-      `https://players.brightcove.net/${encodeURIComponent(account)}` +
-      `/${encodeURIComponent(this.playerId_)}` +
-      `_${encodeURIComponent(embed)}/index.html` +
-      // These are encodeURIComponent'd in encodeId_().
-      (el.getAttribute('data-playlist-id')
-        ? '?playlistId=' + this.encodeId_(el.getAttribute('data-playlist-id'))
-        : el.getAttribute('data-video-id')
-        ? '?videoId=' + this.encodeId_(el.getAttribute('data-video-id'))
-        : '');
-
-    const customReferrer = el.getAttribute('data-referrer');
-
-    if (customReferrer) {
-      el.setAttribute(
-        'data-param-referrer',
-        this.urlReplacements_.expandUrlSync(customReferrer)
-      );
-    }
+    const consentPromise = this.getConsentState();
+    const consentPolicyId = super.getConsentPolicy();
+    const consentStringPromise = consentPolicyId
+      ? getConsentPolicyInfo(this.element, consentPolicyId)
+      : Promise.resolve();
+    const sharedDataPromise = consentPolicyId
+      ? getConsentPolicySharedData(this.element, consentPolicyId)
+      : Promise.resolve();
+
+    return Promise.all([
+      consentPromise,
+      sharedDataPromise,
+      consentStringPromise,
+    ]).then((consents) => {
```

Comment:
Let's move this all to `getConsentState_()` (and rename it to something like `getConsents_()`), then have that return the array of consents, so that too much work doesn't get jammed in this method.

Reply:
thanks for the review, moved in the latest commit